c++/src/kj/async-prelude.h

```diff
@@ -28,23 +28,17 @@
 #include <kj/tuple.h>
 #include <kj/source-location.h>
 
-// Detect whether or not we should enable kj::Promise<T> coroutine integration.
-//
-// TODO(someday): Support coroutines with -fno-exceptions.
-#if !KJ_NO_EXCEPTIONS
-#ifdef __has_include
+// Probe for C++20 or Coroutines TS coroutines.
 #if (__cpp_impl_coroutine >= 201902L) && __has_include(<coroutine>)
 // C++20 Coroutines detected.
 #include <coroutine>
-#define KJ_HAS_COROUTINE 1
 #define KJ_COROUTINE_STD_NAMESPACE std
 #elif (__cpp_coroutines >= 201703L) && __has_include(<experimental/coroutine>)
 // Coroutines TS detected.
 #include <experimental/coroutine>
-#define KJ_HAS_COROUTINE 1
 #define KJ_COROUTINE_STD_NAMESPACE std::experimental
-#endif
-#endif
+#else
+#error "Cap'n Proto requires C++20 or Coroutines TS support"
```

Comment:
This reads a little weird, since we require C++20 either way. Is it meant to be like... "Cap'n Proto requires support for C++20 coroutines. Support for the coroutines TS will suffice."?

Reply:
Fixed!